apis/apiextensions/v1beta1/usage_types.go

```diff
@@ -0,0 +1,114 @@
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// ResourceRef is a reference to a resource.
+type ResourceRef struct {
+	// Name of the referent.
+	Name string `json:"name"`
+}
+
+// ResourceSelector is a selector to a resource.
+type ResourceSelector struct {
+	// MatchLabels ensures an object with matching labels is selected.
+	MatchLabels map[string]string `json:"matchLabels,omitempty"`
+
+	// MatchControllerRef ensures an object with the same controller reference
+	// as the selecting object is selected.
+	MatchControllerRef *bool `json:"matchControllerRef,omitempty"`
+}
+
+// Resource defines a cluster-scoped resource.
+type Resource struct {
+	// API version of the referent.
+	// +optional
+	APIVersion string `json:"apiVersion,omitempty"`
+	// Kind of the referent.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	// +optional
+	Kind string `json:"kind,omitempty"`
+	// Reference to the resource.
+	// +optional
+	ResourceRef *ResourceRef `json:"resourceRef,omitempty"`
+	// Selector to the resource.
+	// This field will be ignored if ResourceRef is set.
+	// +optional
+	ResourceSelector *ResourceSelector `json:"resourceSelector,omitempty"`
+}
+
+// UsageSpec defines the desired state of Usage.
+type UsageSpec struct {
+	// Of is the resource that is "being used".
+	// +kubebuilder:validation:XValidation:rule="has(self.resourceRef) || has(self.resourceSelector)",message="either a resource reference or a resource selector should be set."
+	Of Resource `json:"of"`
+	// By is the resource that is "using the other resource".
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="has(self.resourceRef) || has(self.resourceSelector)",message="either a resource reference or a resource selector should be set."
+	By *Resource `json:"by,omitempty"`
+	// Reason is the reason for blocking deletion of the resource.
+	// +optional
+	Reason *string `json:"reason,omitempty"`
+	// DoNotReplayDeletion will disable deletion replay.
+	// Deletion replay means triggering a deletion on the used resource right
+	// after the usage is deleted, if it was attempted to be deleted at least
+	// once. It is particularly helpful for reducing time for deletion of
+	// the used resource if there were an exponential backoff on the client
+	// attempting to delete the resource, e.g. kube garbage collector.
+	// +optional
+	DoNotReplayDeletion *bool `json:"doNotReplayDeletion,omitempty"`
```

Comment:
I agree this name is awkward.  How about just `replayDeletion` and default to `true`?

Reply:
Yeah that was another possibility but we felt like defaulting to true is not a good practice. 

However, before the naming discussion, I have some observations which made me concerned about defaulting this behavior: https://github.com/crossplane/crossplane/issues/6140#issuecomment-2538629314

At this moment, I am leaning towards keeping it off by default TBH 🤔

Reply:
I can live with it defaulting to `false` since the field is available to be set to `true`.  I think `replayDeletion` is easier to understand.